packages/calypso-e2e/src/lib/blocks/block-flows/payments.ts

```diff
@@ -42,10 +43,11 @@ export class PaymentsBlockFlow implements BlockFlow {
 	 * @param {PublishedPostContext} context The current context for the published post at the point of test execution
 	 */
 	async validateAfterPublish( context: PublishedPostContext ): Promise< void > {
+		const buttonLocator = context.page.locator(
+			`button:has-text("${ this.configurationData.buttonText }")`
+		);
 		// Because Stripe isn't connected to this WordPress.com account, we shouldn't see this block published.
-		if (
-			await context.page.isVisible( `button:has-text("${ this.configurationData.buttonText }")` )
-		) {
+		if ( await buttonLocator.isVisible() ) {
```

Comment:
Note, `locator.isXYZ` methods are borderline deprecated (if not already). See the comment on the `timeout` parameter [here](https://playwright.dev/docs/api/class-locator#locator-is-visible).

I think we might be better off using a `try/catch` with `waitFor` here, or a `count` if steps are taken to wait for the page  to settle beforehand.